### PR TITLE
Remove AssemblyName.CultureName work around

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

**Comment by hishamco** [comment]
After this I think it's the time to fix this work around https://github.com/aspnet/Localization/pull/264 permanently

**Comment by pranavkm** [comment]
Looks good to me :shipit:

File: src/Microsoft.Extensions.Localization/Internal/AssemblyResourceStringProvider.cs [hunk 1]
```diff
@@ -96,11 +95,14 @@ public IList<string> GetAllResourceStrings(CultureInfo culture, bool throwOnMiss
 
         protected virtual AssemblyWrapper GetAssembly(CultureInfo culture)
         {
-            var assemblyString = ApplyCultureToAssembly(culture);
             Assembly assembly;
             try
             {
-                assembly = Assembly.Load(new AssemblyName(assemblyString));
+                var assemblyName = new AssemblyName(_assembly.FullName)
```
Comment (ryanbrandenburg) [inline src/Microsoft.Extensions.Localization/Internal/AssemblyResourceStringProvider.cs]: Move this out of the try-catch, only the code which will throw should be in there.

**Comment by hishamco** [comment]
⌚️   📅

**Comment by hishamco** [comment]
@pranavkm could you please merge this?

**Comment by pranavkm** [comment]
@hishamco - I'll have @ryanbrandenburg merge this in (since it's assigned to him).

**Comment by hishamco** [comment]
Thanks

**Comment by ryanbrandenburg** [comment]
Thanks again @hishamco! Sorry it took me so long to get to this.

**Comment by hishamco** [comment]
No problem @ryanbrandenburg 
thanks

**Pull request merged**